src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -51,35 +55,19 @@ void IWpfTextViewConnectionListener.SubjectBuffersDisconnected(IWpfTextView text
         {
         }
 
-        /// <summary>
-        /// Must be invoked from the UI thread.
-        /// </summary>
-        private void CreateServicesOnUIThread()
+        private async Task InitializeServicesAsync()
         {
-            var serviceTypeAssemblyQualifiedName = typeof(ISnippetInfoService).AssemblyQualifiedName;
-            foreach (var languageService in _languageServices)
-            {
-                if (languageService.Metadata.ServiceType == serviceTypeAssemblyQualifiedName &&
-                    languageService.Metadata.Language == _languageName)
-                {
-                    _ = languageService.Value;
-                    break;
-                }
-            }
-        }
+            await TaskScheduler.Default;
 
-        private void CreateServicesInBackground()
-        {
-            _ = Task.Run(ImportCompletionProviders);
+            var languageServices = _workspace.Services.GetExtendedLanguageServices(_languageName);
+
+            _ = languageServices.GetService<ISnippetInfoService>();
 
             // Preload completion providers on a background thread since assembly loads can be slow
             // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1242321
-            void ImportCompletionProviders()
+            if (languageServices.GetService<CompletionService>() is CompletionServiceWithProviders service)
```

Comment:
with this work, does @genlu need to continue his pr?

Reply:
@genlu already merged the other PR. This is a follow-up to that work.